### PR TITLE
promql: add info() test cases and remove stale XXX comment

### PR DESCRIPTION
**Pull request by aknuds1**
This PR adds test cases for `info()` covering edge cases around non-matching identifying labels and empty-accepting data matchers, and removes a stale XXX comment.

**Commit 1:** Add a test for `info()` where identifying labels don't match any info series, but the data label matcher accepts empty strings (`{data=~".*"}`). The base series should be returned unchanged.

**Commit 2:** Remove the outdated XXX comment about vector selectors requiring at least one non-empty matcher, since the info function's second argument now bypasses this check via `BypassEmptyMatcherCheck`. Replace the workaround test (`{data=~".+", non_existent=~".*"}`) with a simpler test using only an empty-accepting matcher (`{non_existent=~".*"}`).

**Commit 3:** Add a test for `info()` where identifying labels don't match and the data matcher doesn't accept empty strings (`{data=~".+"}`). The series should be dropped since no info series matches and the matcher requires a non-empty value.

#### Does this PR introduce a user-facing change?
```release-notes
NONE
```

### CODE REVIEW

File: promql/promqltest/testdata/info.test [hunk 1]
```diff
@@ -17,6 +17,11 @@ eval range from 0m to 10m step 5m info(metric)
 eval range from 0m to 10m step 5m info(metric_not_matching_target_info)
     metric_not_matching_target_info{instance="a", job="2", label="value"} 0 1 2
 
+# Non-matching identifying labels with a data label matcher that matches empty string.
+# The series should still be returned unchanged.
+eval range from 0m to 10m step 5m info(metric_not_matching_target_info, {data=~".*"})
```
Comment (zenador) [inline promql/promqltest/testdata/info.test:22]: For the sake of completeness, should we also include a test case for `info(metric_not_matching_target_info, {data=~".+"})`?

File: promql/promqltest/testdata/info.test
Comment (zenador) [inline promql/promqltest/testdata/info.test]: I know this wasn't changed in this PR, but I just noticed this, and this limitation doesn't seem to be in effect anymore, so we can update this comment and also have a test case for `info(metric, {non_existent=~".*"})`.

Comment (aknuds1) [reply promql/promqltest/testdata/info.test]: Good suggestion, I've revised the test accordingly.

File: promql/promqltest/testdata/info.test (hunk 1 above)
Comment (aknuds1) [reply promql/promqltest/testdata/info.test:22]: Good suggestion, I've added the test case for completeness.
